Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalPassiveFull.cc

```diff
@@ -0,0 +1,134 @@
+///////////////////////////////////////////////////////////////////////////////
+// File: DDHGCalPassiveFull.cc
+// Description: Geometry factory class for the passive part of a full silicon
+//              module
+// Created by Sunanda Banerjee
+///////////////////////////////////////////////////////////////////////////////
+
+#include <string>
+#include <vector>
+#include <sstream>
+
+#include "DD4hep/DetFactoryHelper.h"
+#include "DataFormats/Math/interface/angle_units.h"
+#include "DetectorDescription/DDCMS/interface/DDPlugins.h"
+#include "DetectorDescription/DDCMS/interface/DDutils.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+//#define EDM_ML_DEBUG
+using namespace angle_units::operators;
+
+struct HGCalPassiveFull {
+  HGCalPassiveFull() { throw cms::Exception("HGCalGeom") << "Wrong initialization to HGCalPassiveFull"; }
+  HGCalPassiveFull(cms::DDParsingContext& ctxt, xml_h e) {
+    cms::DDNamespace ns(ctxt, e, true);
+    cms::DDAlgoArguments args(ctxt, e);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalPassiveFull: Creating an instance";
+#endif
+    std::string parentName = args.parentName();
+    std::string material = args.value<std::string>("ModuleMaterial");
+    double thick = args.value<double>("ModuleThickness");
+    double waferSize = args.value<double>("WaferSize");
+    double waferSepar = args.value<double>("SensorSeparation");
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalPassiveFull: Module " << parentName << " made of " << material << " T "
+                                  << thick << " Wafer 2r " << waferSize << " Half Separation " << waferSepar;
+#endif
+    std::vector<std::string> layerNames = args.value<std::vector<std::string>>("LayerNames");
+    std::vector<std::string> materials = args.value<std::vector<std::string>>("LayerMaterials");
+    std::vector<double> layerThick = args.value<std::vector<double>>("LayerThickness");
+    std::vector<int> copyNumber;
+    copyNumber.resize(materials.size(), 1);
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalPassiveFull: " << layerNames.size() << " types of volumes";
+    for (unsigned int i = 0; i < layerNames.size(); ++i)
+      edm::LogVerbatim("HGCalGeom") << "Volume [" << i << "] " << layerNames[i] << " of thickness " << layerThick[i]
+                                    << " filled with " << materials[i];
+#endif
+    std::vector<int> layerType = args.value<std::vector<int>>("LayerType");
+#ifdef EDM_ML_DEBUG
+    std::ostringstream st1;
+    for (unsigned int i = 0; i < layerType.size(); ++i)
+      st1 << " [" << i << "] " << layerType[i];
+    edm::LogVerbatim("HGCalGeom") << "There are " << layerType.size() << " blocks" << st1.str();
+
+    edm::LogVerbatim("HGCalGeom") << "==>> Executing DDHGCalPassiveFull...";
+#endif
+
+    static constexpr double tol = 0.00001;
```

Comment:
Any reason why this particular number is chosen? Should we not use numeric_limits instead?

Reply:
Like what? This is typically a tolerance accepted by G4 for overlap testing. If you could provide a better one, we can try